azurerm/internal/services/eventgrid/eventgrid_event_subscription_resource.go

```diff
@@ -91,6 +92,30 @@ func resourceArmEventGridEventSubscription() *schema.Resource {
 				Computed: true,
 			},
 
+			"azure_function_endpoint": {
+				Type:          schema.TypeList,
+				MaxItems:      1,
+				Optional:      true,
+				ConflictsWith: utils.RemoveFromStringArray(enpointPropertyNames(), "azure_function_endpoint"),
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
```

Comment:
rather than using `id` (which is vague) can we make this `function_app_id`?

Reply:
Will rename it. But it has to be `function_id`, as the id of the function needs to be used here rather than the one of the function app.